direct_prompt on issue assignment fails with "Error: ASSIGNEE_TRIGGER is required for issue assigned event"

**Describe the bug**
Have been using the normal setup, tagging @claude in issue comments, without any problems.

Trying to add a separate direct_prompt workflow for evaluation of issues on creation and assignment.

Tried initial testing by assigning issues, but I get the following error output:

```
Requesting OIDC token...
Attempt 1 of 3...
OIDC token successfully obtained
Exchanging OIDC token for app token...
Attempt 1 of 3...
App token successfully obtained
Using GITHUB_TOKEN from OIDC
Checking permissions for actor: foobar
Permission level retrieved: admin
Actor has write access: admin
Direct prompt provided, triggering action
Actor type: User
Verified human actor: foobar
✅ Created initial comment with ID: 2932599317
Successfully fetched issue #5352 data
Creating new branch for issue #5352 from source branch: foobar...
Current SHA: 6086fffd4425d50ef6bc2d117f7bc207288e5c7e
From https://github.com/foobar/foobar
 * branch            claude/issue-5352-20250602_213848 -> FETCH_HEAD
 * [new branch]      claude/issue-5352-20250602_213848 -> origin/claude/issue-5352-20250602_213848
Switched to a new branch 'claude/issue-5352-20250602_213848'
M	package-lock.json
branch 'claude/issue-5352-20250602_213848' set up to track 'origin/claude/issue-5352-20250602_213848'.
Successfully created and checked out new branch: claude/issue-5352-20250602_213848
✅ Updated issue comment 2932599317 with branch link
Error: Create prompt failed with error: Error: ASSIGNEE_TRIGGER is required for issue assigned event
Error: Process completed with exit code 1.
```

**To Reproduce**
See example yml below and trigger via issue assignment.

**Expected behavior**
No error output from `anthropics/claude-code-action@beta` with the prompted task being completed.

**Screenshots**
N/A - see workflow logs above

**Workflow yml file**
```yaml
name: Claude Assessment

on:
  issues:
    types:
      - opened
      - reopened
      - assigned

permissions:
  issues: write
  contents: read
  pull-requests: read
  id-token: write

jobs:
  check-for-claude-mentions:
    # ...

  claude-assess-issue:
    needs: check-for-claude-mentions
    if: needs.check-for-claude-mentions.outputs.should-assess == 'true'
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v4
        with:
          fetch-depth: 1

      - name: Set up Node.js
        # ...

      - name: Authenticate with GCP
        # ...

      - name: Install dependencies
        # ...

      - name: Install Google Cloud CLI
        # ...
        # ...

      - name: Download secrets from secrets manager
        working-directory: foobar
        # ...

      - name: Run Claude Assessment
        uses: anthropics/claude-code-action@beta
        with:
          anthropic_api_key: ${{ secrets.ANTHROPIC_API_KEY }}
          model: claude-sonnet-4-20250514
          timeout_minutes: '30'
          allowed_tools: 'Bash(git status),Bash(git log),Bash(git show),Bash(git blame),Bash(git reflog),Bash(git stash list),Bash(git ls-files),Bash(git branch),Bash(git tag),Bash(git diff),View,GlobTool,GrepTool,BatchTool'
          direct_prompt: |
            Please generate a detailed assessment of your ability to solve this issue. This is for evaluation purposes only - do not attempt to implement any solutions or create any branches.

            Confidence levels: Almost Certain, Likely, Possible, Unlikely, Big Nope

            1. Carefully review this issue. How confident are you that you fully understand the issue?
            2. Carefully review the codebase. How confident are you that you fully understand what code paths in which files are relevant to this issue?
            3. How confident are you that you can ultimately solve this issue?

            Please provide a confidence level for each of the above questions, including justification for your determinations and evidence that supports those determinations.

            After the assessment, if your confidence level for solving the issue is "Almost Certain" or "Likely", please provide an implementation plan.

```

**API Provider**

[x] Anthropic First-Party API (default)
[ ] AWS Bedrock
[ ] GCP Vertex

**Additional Context**

It also ignores instructions not to create a branch, not sure if that's a bug or expected behavior.

--- a/src/create-prompt/index.ts
+++ b/src/create-prompt/index.ts
@@ -242,7 +242,7 @@ export function prepareContext(
       }
 
       if (eventAction === "assigned") {
-        if (!assigneeTrigger) {
+        if (!assigneeTrigger && !directPrompt) {
           throw new Error(
             "ASSIGNEE_TRIGGER is required for issue assigned event",
           );
@@ -254,7 +254,7 @@ export function prepareContext(
           issueNumber,
           baseBranch,
           claudeBranch,
-          assigneeTrigger,
+          ...(assigneeTrigger && { assigneeTrigger }),
         };
       } else if (eventAction === "opened") {
         eventData = {
@@ -331,7 +331,9 @@ export function getEventTypeAndContext(envVars: PreparedContext): {
       }
       return {
         eventType: "ISSUE_ASSIGNED",
-        triggerContext: `issue assigned to '${eventData.assigneeTrigger}'`,
+        triggerContext: eventData.assigneeTrigger
+          ? `issue assigned to '${eventData.assigneeTrigger}'`
+          : `issue assigned event`,
       };
 
     case "pull_request":

--- a/src/create-prompt/types.ts
+++ b/src/create-prompt/types.ts
@@ -65,7 +65,7 @@ type IssueAssignedEvent = {
   issueNumber: string;
   baseBranch: string;
   claudeBranch: string;
-  assigneeTrigger: string;
+  assigneeTrigger?: string;
 };
 
 type PullRequestEvent = {

--- a/test/create-prompt.test.ts
+++ b/test/create-prompt.test.ts
@@ -614,6 +614,29 @@ describe("getEventTypeAndContext", () => {
     expect(result.eventType).toBe("ISSUE_ASSIGNED");
     expect(result.triggerContext).toBe("issue assigned to 'claude-bot'");
   });
+
+  test("should return correct type and context for issue assigned without assigneeTrigger", () => {
+    const envVars: PreparedContext = {
+      repository: "owner/repo",
+      claudeCommentId: "12345",
+      triggerPhrase: "@claude",
+      directPrompt: "Please assess this issue",
+      eventData: {
+        eventName: "issues",
+        eventAction: "assigned",
+        isPR: false,
+        issueNumber: "999",
+        baseBranch: "main",
+        claudeBranch: "claude/issue-999-20240101_120000",
+        // No assigneeTrigger when using directPrompt
+      },
+    };
+
+    const result = getEventTypeAndContext(envVars);
+
+    expect(result.eventType).toBe("ISSUE_ASSIGNED");
+    expect(result.triggerContext).toBe("issue assigned event");
+  });
 });
 
 describe("buildAllowedToolsString", () => {

--- a/test/prepare-context.test.ts
+++ b/test/prepare-context.test.ts
@@ -219,6 +219,55 @@ describe("parseEnvVarsWithContext", () => {
         ),
       ).toThrow("BASE_BRANCH is required for issues event");
     });
+
+    test("should allow issue assigned event with direct_prompt and no assigneeTrigger", () => {
+      const contextWithDirectPrompt = createMockContext({
+        ...mockIssueAssignedContext,
+        inputs: {
+          ...mockIssueAssignedContext.inputs,
+          assigneeTrigger: "", // No assignee trigger
+          directPrompt: "Please assess this issue", // But direct prompt is provided
+        },
+      });
+
+      const result = prepareContext(
+        contextWithDirectPrompt,
+        "12345",
+        "main",
+        "claude/issue-123-20240101_120000",
+      );
+
+      expect(result.eventData.eventName).toBe("issues");
+      expect(result.eventData.isPR).toBe(false);
+      expect(result.directPrompt).toBe("Please assess this issue");
+      if (
+        result.eventData.eventName === "issues" &&
+        result.eventData.eventAction === "assigned"
+      ) {
+        expect(result.eventData.issueNumber).toBe("123");
+        expect(result.eventData.assigneeTrigger).toBeUndefined();
+      }
+    });
+
+    test("should throw error when neither assigneeTrigger nor directPrompt provided for issue assigned event", () => {
+      const contextWithoutTriggers = createMockContext({
+        ...mockIssueAssignedContext,
+        inputs: {
+          ...mockIssueAssignedContext.inputs,
+          assigneeTrigger: "", // No assignee trigger
+          directPrompt: "", // No direct prompt
+        },
+      });
+
+      expect(() =>
+        prepareContext(
+          contextWithoutTriggers,
+          "12345",
+          "main",
+          "claude/issue-123-20240101_120000",
+        ),
+      ).toThrow("ASSIGNEE_TRIGGER is required for issue assigned event");
+    });
   });
 
   describe("optional fields", () => {

